jconstraints-core/src/main/java/gov/nasa/jpf/constraints/expressions/IfThenElse.java

```diff
@@ -59,7 +61,19 @@ public E evaluate(Valuation values) {
 
   @Override
   public E evaluateSMT(Valuation values) {
-    return ifCond.evaluateSMT(values) ? thenExpr.evaluateSMT(values) : elseExpr.evaluateSMT(values);
+    boolean condition;
+    try {
+      condition = ifCond.evaluateSMT(values);
+    } catch (UndecidedBooleanExeception e1) {
+      try {
+        E thenV = thenExpr.evaluateSMT(values);
+        E elseV = elseExpr.evaluateSMT(values);
+        throw new UndecidedIfException(thenV, elseV);
+      } catch (UndecidedBooleanExeception e2) {
+        throw e1;
```

Comment:
This silently discards `e2`. Java 1.7 introduced a mechanism to report this:
```suggestion
        e1.addSuppressed(e2);
        throw e1;
```